### ISSUE TITLE
Bulkloader: incorrect parsing of complex identifiers

### ISSUE THREAD

**Issue by camwebb** [ArctosDB/arctos#4723]
**Describe the bug**

Complex identifiers (e.g. 'JRF14-048') are incorrectly parsed by the bulkloader. In this [example](https://arctos.database.museum/guid/UAM:Herb:256663), the parsed result has _prefix_ = JRF14-048 and the _integer part_ = 48, leading to an (incorrect) recreated identifier of `JRF14-04848`.

**To Reproduce**

1. Upload specimen record to Bulkloader with `other_id_num_1` = 'JRF14-048' and `other_id_num_type_1` = 'collector number'
2. Successfully load 
3. Check via 'Other IDs' tab in specimen record, or via SQL `SELECT * from flat where othercatalognumbers like '%V178740%'`

**Expected behavior**

Prefix = 'JRF14-048' and integer part = '' ? I understand why this is hard to parse!

**Comment by dustymc** [comment]
Thanks - SOMEONE had a cryptically-named variable, and the 'stuff going out put back together looks like what came in' check was - failing at failing, I guess...

FYI the script generally should stuff everything into prefix if there's any ambiguity:

```
arctosprod@arctos>>     select split_other_id('JRF14-048') ;
                   split_other_id                   
----------------------------------------------------
 {"prefix":"JRF14-048","number":null,"suffix":null}
```
But you can specify if something makes sense to humans - https://handbook.arctosdb.org/documentation/other-identifying-numbers.html#bulk-loading-rules.

```
arctosprod@arctos>>     select split_other_id('JRF[14]-048') ;
                split_other_id                
----------------------------------------------
 {"prefix":"JRF","number":14,"suffix":"-048"}
```

Tentatively closing, but let me know if you need any help cleaning up or whatever.
